libdrgn/debug_info.c

```diff
@@ -3446,7 +3446,42 @@ drgn_debug_info_find_complete(struct drgn_debug_info *dbinfo, uint64_t tag,
 	struct drgn_dwarf_index_die *index_die =
 		drgn_dwarf_index_iterator_next(&it);
 	if (!index_die)
-		return &drgn_not_found;
+	{
+		Dwarf_Die *ancestors = NULL;
+		size_t ancestors_len;
+		if ((err = drgn_find_die_ancestors(incomplete_die, &ancestors, &ancestors_len)))
+			goto err;
+
+		struct drgn_dwarf_index_namespace *ns = &dbinfo->dindex.global;
+		uint64_t ns_tag = DW_TAG_namespace;
+		for (int i = 0; i < ancestors_len; i++) {
+			if (dwarf_tag(&ancestors[i]) != DW_TAG_namespace)
+				continue;
+			const char *name = dwarf_diename(&ancestors[i]);
```

Comment:
This can be `NULL` if the namespace is unnamed. I don't think we index unnamed namespaces (although we definitely should), but for now, I think you should just skip over unnamed namespaces, since their contents should technically be found in the enclosing namespace.